### DISCUSSION TITLE
ArgoCD manual configuration

### DISCUSSION THREAD

**Discussion by c0c0n3** [c0c0n3/kitt4sme.live#146] [Q&A]
Question originally asked by @etoschi in the [old forum](https://github.com/c0c0n3/kitt4sme/discussions/12).

How do I check if ArgoCD is correctly configured? I guess that I could “manually” setup connection to the git repository through the ArgoCD UI but I wonder why these manual config steps are not mentioned in the bootstrap guide and for this reason I guess this is not the correct procedure.

I also noticed that while configuring the repository URL in custom-urls.yaml, there’s no mention to configuring credentials to access this repository (username and password or ssh key), am I missing something?

From the UI, if I try to define a new connection via Repositories > CONNECT REPO USING SSH/HTTPS, what is the right value to select for “Project”?

 
![image002](https://user-images.githubusercontent.com/101259854/166196746-813927de-1f8e-4cb4-83af-6c094ffa723b.png)

**Comment by c0c0n3** [comment]
@etoschi thanks for bringing this up!

> How do I check if ArgoCD is correctly configured?

You can refer to the ArgoCD manual depending on what you need to check. But if you can login and see all the projects and apps there with no issues in the UI (red bars) then your config should be pretty much okay I'd guess.

 > I could “manually” setup connection to the git repository through the ArgoCD UI but I wonder why these manual config steps are not mentioned in the bootstrap guide and for this reason I guess this is not the correct procedure.

Yes you can, but then you should delete the [ArgoCD directory](https://github.com/c0c0n3/kitt4sme.live/tree/main/deployment/mesh-infra/argocd) from your fork. In fact, we've adopted a GitOps approach where we keep absolutely everything in config files in the repo to ensure reproducibility, etc. That's why manually configuring ArgoCD isn't mentioned in the bootstrap procedure.

> I also noticed that while configuring the repository URL in custom-urls.yaml, there’s no mention to configuring credentials to access this repository (username and password or ssh key), am I missing something?

No you're right. At the moment the ArgoCD config we have works with a public repo. But you should be able to change the [base config](https://github.com/c0c0n3/kitt4sme.live/blob/main/deployment/mesh-infra/argocd/projects/base/app.yaml) to specify credentials for a private repo as explained in the ArgoCD manual.

> From the UI, if I try to define a new connection via Repositories > CONNECT REPO USING SSH/HTTPS, what is the right value to select for “Project”?

If you go for a manual configuration (i.e. you zap the ArgoCD dir in your fork like I said earlier), then you can enter whichever project name you fancy. If you take the GitOps approach (recommended!), then you should edit the [base config](https://github.com/c0c0n3/kitt4sme.live/blob/main/deployment/mesh-infra/argocd/projects/base/app.yaml) instead. But the best option is probably to keep our default settings: we've got a project in correspondence of each platform layer---mesh infra, plat infra services, and app services.

**Comment by c0c0n3** [comment]
Reposting @etoschi's reply from the old forum [thread][reply].

Ok, but I have a doubt it’s taking the configuration from the wrong place: I mean from [github.com/c0c0n3/kitt4sme.live](http://github.com/c0c0n3/kitt4sme.live) instead of the url I changed in custom-urls.yaml; this poses a “chicken and egg” like problem, since (a) I must make changes yamls on my private repository to change configuration for ArgoCD and  (b) until ArgoCD isn’t correctly configured, it won’t take changes I make to configuration in yamls. I wonder If I followed some steps in the guide wrongly, and have to specify the corporate git repository URL somewhere else beyond the custom-urls.yaml


[reply]: https://github.com/c0c0n3/kitt4sme/discussions/12#discussioncomment-2688275

**Comment by c0c0n3** [comment]
> but I have a doubt it’s taking the configuration from the wrong place: I mean from [github.com/c0c0n3/kitt4sme.live](https://github.com/c0c0n3/kitt4sme.live) instead of the url I changed in custom-urls.yaml

There's an easy way to tell. Just check out your fork locally, install Nix as explained in the bootstrap procedure and then run, in your root repo dir:

```console
$ cd nix
$ nix shell
$ cd ../deployment/mesh-infra/
$ kustomize build | grep repoURL
```

The URLs you see should be those of your repo, not ours. E.g. this is the output I get on my side

```
repoURL: https://github.com/c0c0n3/kitt4sme.live
repoURL: https://github.com/c0c0n3/kitt4sme.live
repoURL: https://github.com/c0c0n3/kitt4sme.live
...
```

For you the URL should be that you entered in `custom-urls.yaml` or if you're not using `custom-urls.yaml`, the URL you entered in the [base config](https://github.com/c0c0n3/kitt4sme.live/blob/main/deployment/mesh-infra/argocd/projects/base/app.yaml) as explained earlier.

> “chicken and egg” like problem

In principle there should be no chicken-and-egg problem, if there is, it's our fault :-) In fact, if you do this:

1. Fork our repo.
2. Edit your fork to change the ArgoCD repo URL, push your changes to your fork.
3. Install the cluster from your fork.

Then when ArgoCD gets deployed in your cluster, there should be an ArgoCD app (K8s resource) for each cluster component and the URL of each app resource should point to your fork. To build the cluster, your ArgoCD service will read those app resources from K8s, so the configuration it'll use in the end is that coming from your repo.

Let us know how if you can move past this issue...

**Reply by cmelchiorre** [reply]
actually the problem seems to be with the kustomize build, since I still get the c0c0n3 url in the grep.

In any case since your last reply I already found the workaround of changing the URL directly in the base config, instead of using the overrides in custom-urls. Doing so, I managed to make the workflow work (ie. push my app's descriptor and observe the deployment taking place on the cluster via ArgoCD UI), so this is no longer a blocking issue for me (thanks in any case).

(By the way, I had to revert to use the fork on github/cmelchiorre as origin, instead of the actual corporate gitlab repo, due to some problems in authenticating the cluster with the gitlab, but this is a minor problem).

**Reply by c0c0n3** [reply]
hi @cmelchiorre :-)

> still get the c0c0n3 url in the grep

that's most likely a bug :-) track #145 for progress on that front.

> changing the URL directly in the base config

excellent!

> problems in authenticating the cluster with the gitlab

yeah, we don't support private repos at the moment, but you can easily customise our ArgoCD config to do so. Get in touch if you need this feature going forward, we'll make a plan.
